### PR TITLE
fix(GC): removing Genesis Block from State

### PR DESCRIPTION
**Pull request by Kouprin**
Currently, we're not using any data from Genesis Block in State. According to https://github.com/nearprotocol/nearcore/issues/2570, it's not enough to not erase Genesis data from State to keep it useful. So, it's better to remove Genesis data from State at all.

Fixes https://github.com/nearprotocol/nearcore/issues/2570

Plan to test: CI + sanity python tests (gc_after_sync, gc_sync_after_sync, gc_sync_after_sync swap_nodes).

### CODE REVIEW

**Comment by mikhailOK** [comment]
We need better test coverage then - right now all green.
Can genesis state be garbage collected? Currently it is.

**Comment by Kouprin** [comment]
@bowenwang1996 @mikhailOK that's right. Which components of the system needs Genesis Block?

**Comment by bowenwang1996** [comment]
@Kouprin actually genesis block is always reconstructable. However, for the purposes of external services like explorer, it is better to keep genesis block.

**Comment by Kouprin** [comment]
UPD: keep Genesis Block alive. Genesis is removed from State only.

File: chain/chain/src/store.rs [hunk 1]
```diff
@@ -1791,6 +1782,7 @@ impl<'a> ChainStoreUpdate<'a> {
                 // Broken GC prerequisites found
                 assert!(false);
             }
+            // Don't clean Genesis Block
             return Ok(());
```
Comment (mikhailOK) [inline chain/chain/src/store.rs:1786]: `store_update` never gets committed if we just return

Comment (Kouprin) [reply chain/chain/src/store.rs:1786]: I wonder which tests can help us to catch this types of bugs related to Genesis?

Comment (mikhailOK) [reply chain/chain/src/store.rs:1787]: we need a storage consistency check.
collect all records from all columns and check for leaks and dangling references

Comment (Kouprin) [reply chain/chain/src/store.rs:1787]: It seems it's an Epic-size test.
Definitely should be done.

**Pull request merged**